openwisp_network_topology/base/link.py

```diff
@@ -99,6 +99,33 @@ def validate_topology(self):
         if errors:
             raise ValidationError(errors)
 
+    def validate_organization(self):
+        if self.topology.organization_id is None:
+            # Shared link is only created between nodes of
+            # two different organizations.
+            if self.source.organization_id == self.target.organization_id:
+                self.organization_id = self.source.organization_id
+            else:
+                self.organization_id = None
+        else:
+            if self.source.organization_id != self.topology.organization_id:
+                raise ValidationError(
+                    {
+                        'source': _(
+                            'Source node and topology should have same organization.'
+                        )
+                    }
+                )
+            if self.target.organization_id != self.topology.organization_id:
+                raise ValidationError(
+                    {
+                        'target': _(
+                            'Target node and topology should have same organization.'
+                        )
+                    }
+                )
```

Comment:
Please let's use the dict trick as I have shown a few weeks ago, let's make this a habit (so that if both conditions trigger, both validation errors are shown and not only the first one).